og_access/src/Plugin/OgFields/OgAccessField.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace Drupal\og_access\Plugin\OgFields;
+
+use Drupal\og\OgFieldBase;
+use Drupal\og\OgFieldsInterface;
+
+/**
+ * Determine if group should use default roles and permissions.
+ *
+ * @OgFields(
+ *  id = \Drupal\og_access\OgAccess::OG_ACCESS_FIELD,
+ *  type = "group",
+ *  description = @Translation("Determine access to the group.")
+ * )
+ */
+class OgAccessField extends OgFieldBase implements OgFieldsInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFieldStorageBaseDefinition(array $values = []) {
+    $values += [
+      'cardinality' => 1,
+      'settings' => [
+        'allowed_values' => [
+          0 => $this->t('Public - accessible to all site users'),
```

Comment:
See the remark in #644.

Reply:
Ok, I was not sure whether to put my observation in the og_access or the og_migration branch. Will this be the access structure and should migration cope with that? Or will access resemble the D7 behaviour in which case the migration will be straightforward?

Reply:
You are right that it has changed between d7 and d8, I don't know the history of this PR but I'm guessing it's already being used by some.

I prefer solving this in the migration rather than here.
@pfrenssen, @amitaibu, @zerolab, your thoughts on this are welcome.

Reply:
Answering #644 here:

The reason for `0|Use group defaults` -- is that we'd have a config (in d7 - `variable_get`) -- that would indicate the default value to use.

I think removing this in favor of explicit set for each group (i.e. `public` or `private`) is fine. For two reasons:

1. IMO being explicit is better when it comes to access
1. If switching config from public to private, we'd have to iterate over all those groups and re-build their access. So without this config, is one less thing to care about.

Reply:
@amitaibu  It is not clear what you suggest  for existing groups in D7. Is the Migration going to take care of this to keep private groups private?

Reply:
Manual correction:
```
UPDATE node_access na
INNER JOIN node__group_content_access gca on na.nid = gca.entity_id
SET na.realm = 'og_access:node'
WHERE gca.group_content_access_value = 2
```

(Be sure to do a cache clear afterwards)